models/articles.models.js

```diff
@@ -51,8 +51,54 @@ exports.getTheseComments = (id) => {
     })
 } 
 
-exports.readAllArticles = () => {
-    return db.query('SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.body, articles. created_at, articles.votes, CAST (COUNT (*) AS INT) AS comment_count FROM articles JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id')
+exports.readAllArticles = (query) => {
+    let queryStr = "SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.body, articles. created_at, articles.votes, CAST (COUNT (*) AS INT) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id"
+
+    //handles 'topic' filter query first
+    const topicArr = []
+    if (query.topic) {
+        queryStr += ' WHERE articles.topic = $1'
+        topicArr.push(query.topic)
+    }
+
+    //group by
+    queryStr += ' GROUP BY articles.article_id'
+
+    //handles sort_by query
+    const greenList = ['title', 'created_at', 'author', 'topic', 'votes', 'body']
+
+    if(query.sort_by) {
+        if (!greenList.includes(query.sort_by)) {
+            return Promise.reject({status:400, msg: 'invalid query'})
+        } else {
+            queryStr += ` ORDER BY articles.${query.sort_by}`
+        }
+
+        queryStr += query.sort_by == 'votes' ? ' DESC' : ' ASC';
+
+    } else {
+        queryStr += ' ORDER BY articles.created_at DESC'
+    }
+
+
+    //handles 'order' query
+    const orderGreenList = ['ASC', 'DESC']
+    
+    if(query.order) {
```

Comment:
This block feels like unnecessary and not very dry code. We shouldnt be creating a string, only to later edit it. We should be building the string up. If you have this here, it means you are doing it in the wrong order. Try to refactor so we can do things in the right order without having to use .replace on the string. We shouldnt have to replace anything on a string that you are creating right here and now, and have full control over